# Inspección inicial del dataset
## Proyecto: Factores para el éxito comercial en la industria de videojuegos para orientar la inversión en nuevos proyectos.

**Equipo:** Risk Analysts

**Alumno:** Hernández Martínez Ricardo

**Fecha:** 16 de septiembre de 2026

In [4]:
import pandas as pd

# Dataset principal
df = pd.read_csv('vgsales.csv')

# Fuente alternativa
df_alt = pd.read_csv('videoGameSalesWithRatings.csv')

print("Datasets cargados correctamente.")
print(f"Dataset principal -> Filas: {df.shape[0]}, Columnas: {df.shape[1]}")
print(f"Fuente alternativa -> Filas: {df_alt.shape[0]}, Columnas: {df_alt.shape[1]}")

Datasets cargados correctamente.
Dataset principal -> Filas: 16598, Columnas: 11
Fuente alternativa -> Filas: 16719, Columnas: 16


In [5]:
# Mostrar las primeras 10 filas del dataset principal
df.head(10)

,Rank,Name,Platform,Year,Genre,Publisher,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
0,1,Wii Sports,Wii,2006.0,Sports,Nintendo,41.49,29.02,3.77,8.46,82.74
1,2,Super Mario Bros.,NES,1985.0,Platform,Nintendo,29.08,3.58,6.81,0.77,40.24
2,3,Mario Kart Wii,Wii,2008.0,Racing,Nintendo,15.85,12.88,3.79,3.31,35.82
3,4,Wii Sports Resort,Wii,2009.0,Sports,Nintendo,15.75,11.01,3.28,2.96,33.00
4,5,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,Nintendo,11.27,8.89,10.22,1.00,31.37
5,6,Tetris,GB,1989.0,Puzzle,Nintendo,23.20,2.26,4.22,0.58,30.26
6,7,New Super Mario Bros.,DS,2006.0,Platform,Nintendo,11.38,9.23,6.50,2.90,30.01
7,8,Wii Play,Wii,2006.0,Misc,Nintendo,14.03,9.20,2.93,2.85,29.02
8,9,New Super Mario Bros. Wii,Wii,2009.0,Platform,Nintendo,14.59,7.06,4.70,2.26,28.62
9,10,Duck Hunt,NES,1984.0,Shooter,Nintendo,26.93,0.63,0.28,0.47,28.31


In [6]:
# Información general: tipos de datos y valores no nulos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


In [8]:
# Lista de columnas del dataset principal
print("Columnas del dataset vgsales.csv:")
for col in df.columns:
    print(f"- {col}")

Columnas del dataset vgsales.csv:
- Rank
- Name
- Platform
- Year
- Genre
- Publisher
- NA_Sales
- EU_Sales
- JP_Sales
- Other_Sales
- Global_Sales


In [9]:
# Conteo de valores faltantes por columna
print("Valores faltantes por columna (dataset vgsales.csv):")
print(df.isnull().sum())
print("\nPorcentaje de faltantes:")
print((df.isnull().sum() / len(df)) * 100)

Valores faltantes por columna (dataset vgsales.csv):
Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

Porcentaje de faltantes:
Rank            0.000000
Name            0.000000
Platform        0.000000
Year            1.632727
Genre           0.000000
Publisher       0.349440
NA_Sales        0.000000
EU_Sales        0.000000
JP_Sales        0.000000
Other_Sales     0.000000
Global_Sales    0.000000
dtype: float64


In [ ]:
# Estadísticas descriptivas de columnas numéricas
df.describe()

,Rank,Year,NA_Sales,EU_Sales,JP_Sales,Other_Sales,Global_Sales
count,16598.000000,16327.000000,16598.000000,16598.000000,16598.000000,16598.000000,16598.000000
mean,8300.605254,2006.406443,0.264667,0.146652,0.077782,0.048063,0.537441
std,4791.853933,5.828981,0.816683,0.505351,0.309291,0.188588,1.555028
min,1.000000,1980.000000,0.000000,0.000000,0.000000,0.000000,0.010000
25%,4151.250000,2003.000000,0.000000,0.000000,0.000000,0.000000,0.060000
50%,8300.500000,2007.000000,0.080000,0.020000,0.000000,0.010000,0.170000
75%,12449.750000,2010.000000,0.240000,0.110000,0.040000,0.040000,0.470000
max,16600.000000,2020.000000,41.490000,29.020000,10.220000,10.570000,82.740000


In [10]:
# Conteo de filas duplicadas
duplicados = df.duplicated().sum()
print(f"Filas duplicadas (dataset vgsales.csv): {duplicados}")

Filas duplicadas (dataset vgsales.csv): 0


In [ ]:
# Valores únicos en columnas categóricas
print("Plataformas únicas:", df['Platform'].nunique())
print("Géneros únicos:", df['Genre'].nunique())
print("Publishers únicos:", df['Publisher'].nunique())
print("\nGéneros disponibles:")
print(df['Genre'].value_counts())

Plataformas únicas: 31
Géneros únicos: 12
Publishers únicos: 578

Géneros disponibles:
Genre
Action          3316
Sports          2346
Misc            1739
Role-Playing    1488
Shooter         1310
Adventure       1286
Racing          1249
Platform         886
Simulation       867
Fighting         848
Strategy         681
Puzzle           582
Name: count, dtype: int64


In [ ]:
# Rango de años cubiertos
print(f"Año mínimo: {df['Year'].min()}")
print(f"Año máximo: {df['Year'].max()}")
print(f"Registros sin año: {df['Year'].isnull().sum()}")

Año mínimo: 1980.0
Año máximo: 2020.0
Registros sin año: 271


In [12]:
# Información de la fuente alternativa
print("=== videoGameSalesWithRatings.csv ===")
print(f"Filas: {df_alt.shape[0]}, Columnas: {df_alt.shape[1]}")
print("\nColumnas:")
for col in df_alt.columns:
    print(f"- {col}")
print("\nValores faltantes:")
print(df_alt.isnull().sum())

=== videoGameSalesWithRatings.csv ===
Filas: 16719, Columnas: 16

Columnas:
- Name
- Platform
- Year_of_Release
- Genre
- Publisher
- NA_Sales
- EU_Sales
- JP_Sales
- Other_Sales
- Global_Sales
- Critic_Score
- Critic_Count
- User_Score
- User_Count
- Developer
- Rating

Valores faltantes:
Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         6704
User_Count         9129
Developer          6623
Rating             6769
dtype: int64


## Conclusiones

### Dataset principal (vgsales.csv)
- Tiene 16,598 filas y 11 columnas.
- La unidad de observación es un videojuego.
- Columnas principales: Name, Platform, Year, Genre, Publisher y ventas por región.
- Problemas visibles: valores faltantes en Year y Publisher.
- No se detectaron filas duplicadas.
- Los datos llegan hasta 2020.

### Fuente alternativa (videoGameSalesWithRatings.csv)
- Tiene 16,719 filas y 16 columnas.
- Incluye calificaciones de crítica y usuarios (Critic_Score, User_Score).
- Problemas visibles: muchos valores faltantes en columnas de ratings.
- Algunos valores de User_Score son "tbd" en lugar de números.
- Los datos llegan hasta 2017.

### Próximos pasos
- Documentar los faltantes y decidir su tratamiento en la etapa de limpieza.
- Estandarizar nombres de columnas entre ambos datasets.
- Cargar los datos limpios en PostgreSQL para las consultas SQL.